homeassistant/components/linkplay/__init__.py

```diff
@@ -26,13 +26,20 @@ class LinkPlayData:
 
 type LinkPlayConfigEntry = ConfigEntry[LinkPlayData]
 
+async def options_update_listener(hass: HomeAssistant, entry: LinkPlayConfigEntry):
+    """Handle options update."""
+    await hass.config_entries.async_reload(entry.entry_id)
+
 
 async def async_setup_entry(hass: HomeAssistant, entry: LinkPlayConfigEntry) -> bool:
     """Async setup hass config entry. Called when an entry has been setup."""
 
     session: ClientSession = await async_get_client_session(hass)
     bridge: LinkPlayBridge | None = None
 
+    # Register update listener to update config entry when options are updated.
+    entry.async_on_unload(entry.add_update_listener(options_update_listener))
+
```

Comment:
This needs to be at the bottom of the function, to ensure it only unloads when it was successfully setup.